packages/honeycomb-opentelemetry-web/examples/experimental/user-interaction-instrumentation/src/reactRouterSpanProcessor.ts

```diff
@@ -0,0 +1,48 @@
+import { SpanProcessor } from '@opentelemetry/sdk-trace-base';
+import { Span } from '@opentelemetry/api';
+
+/**
+ * SpanProcessor that adds attributes to spans based on the state of the React Router
+ * Sets the page.route attribute to the generic dynamic route
+ * Records the span as an error if there are errors in the router state (e.g. 404)
+ */
+export class ReactRouterSpanProcessor implements SpanProcessor {
```

Comment:
Should this live in this example app? Or should we ship it from our distro?

Reply:
GREAT Q! I think we can leave that as a followup discussion, since I don't think it the exports for the package have changes, unless I missed something?

Reply:
i just meant that this looks like a useful span processor that we could ship for our clients to use

Reply:
💯 Let's follow up and make that call!